src/agent/src/device.rs

```diff
@@ -401,6 +401,48 @@ async fn get_vfio_device_name(sandbox: &Arc<Mutex<Sandbox>>, grp: IommuGroup) ->
     Ok(format!("{}/{}", SYSTEM_DEV_PATH, &uev.devname))
 }
 
+#[derive(Debug)]
+struct MmioBlockMatcher {
+    suffix: String,
```

Comment:
- ooi, why can't we match on the entire udev path?
- Could you add some unit tests using fake `Uevent` events? That would help understanding and also clarify the point above.

Reply:
Normally, there are two path for a device, it could be `sys/block/vda`, and `sys/devices/ ......./virtio0/block/vda`, the first one will point the second one, so match the suffix would be enough.